### PR TITLE
Raycaster: Deprecate .linePrecision.

### LINKED ISSUE #18590: Inconsistent raycast precision API
##### Description of the problem

Both `Points` and `Line` implement a `raycast()` function/method. They both use a threshold/precision:

https://github.com/mrdoob/three.js/blob/138d25dbd138f9feedbb20ade26f104155a50d8f/src/objects/Points.js#L37-L59

https://github.com/mrdoob/three.js/blob/138d25dbd138f9feedbb20ade26f104155a50d8f/src/objects/Line.js#L94-L117

I think the API is a bit inconsistent and they could both be named either threshold or precision and also they could both be set in the `Raycaster` in the same way. Right now you have to:

```javascript
raycaster.linePrecision = 42;
raycaster.params.Points.threshold = 42;
```

Also, only `linePrecision` [is documented](https://threejs.org/docs/index.html#api/en/core/Raycaster.linePrecision).

##### Proposal

Same API for all thresholds/precisions. It seems the `.params` attribute is more complete/flexible, so I would say:

```javascript
raycaster.params.Line.threshold = 42;
raycaster.params.Points.threshold = 42;
```

That means removing the `linePrecision` attribute.

##### Three.js version

- [x] Dev
- [x] r113

### PR DESCRIPTION
**Pull request by Peque**
Fixes #18590.

~There were no previous tests for Raycaster's precision/threshold, so I added none myself. :sweat_smile:~

First PR in this project and very little experience with JS. Review with care... :innocent:

### CODE REVIEW

**Comment by Peque** [comment]
I ended-up adding some tests.

File: test/unit/src/core/Raycaster.tests.js
Comment (Mugen87) [inline test/unit/src/core/Raycaster.tests.js]: Please avoid the usage of `Geometry`. `BufferGeometry` should be your first choice. Use this pattern:
```js
var points = [ new Vector3( -2, -10, -5 ), new Vector3( -2, 10, -5 ) ];
var geometry = new BufferGeometry().setFromPoints( points );
```

**Comment by Mugen87** [comment]
Please add this code to [Three.Legacy.js](https://github.com/mrdoob/three.js/blob/dev/src/Three.Legacy.js), otherwise use code will break.
```js
Object.defineProperties( Raycaster.prototype, {

	linePrecision: {
		get: function () {

			console.warn( 'THREE.Raycaster: .linePrecision has been deprecated. Use .params.Line.threshold instead.' );
			return this.params.Line.threshold;

		},
		set: function ( value ) {

			console.warn( 'THREE.Raycaster: .linePrecision has been deprecated. Use .params.Line.threshold instead.' );
			this.params.Line.threshold = value;

		}
	}

} );
```

File: test/unit/src/core/Raycaster.tests.js
Comment (Peque) [reply test/unit/src/core/Raycaster.tests.js]: Fixed in 101dc18ef4a691ad77ecaf0045ffd00ed8d3418d.

Definitely more compact. :blush:

I think I just got the idea of using `Geometry` looking for how other `Line`s where created in `test/unit/src/core/`, and found just one example:

https://github.com/mrdoob/three.js/blob/fd8b70cf67f4a1ee49817b13bc2deb59f5369fb6/test/unit/src/core/BufferGeometry.tests.js#L364-L377

**Comment by Peque** [comment]
@Mugen87 Added the deprecation warning in b30437b42df1ddc78b44ae2b13866519db5a65b9. Seems to be working fine:

Screenshot from 2020-02-17 12-44-33 

I can squash the fixup commits before merging. Feel free to squash them yourself if you prefer so.

I see you changed the MR description, so maybe you want me to change the commit subject too?

**Comment by Mugen87** [comment]
> I see you changed the MR description, so maybe you want me to change the commit subject too?

No, not necessary. I've only changed the PR title since it makes my life easier when writing the release notes^^.

**Review by Mugen87 — APPROVED** [review]

**Comment by Peque** [comment]
> No, not necessary. I've only changed the PR title since it makes my life easier when writing the release notes^^.

@Mugen87  Understood. :blush: :+1: 

Do you want me to squash the fixup commits and force-push the changes? I think GitHub allows you to see diffs between force-pushed changes (to verify I did not introduce new changes since your last review), but if you want to squash them yourself, I'll leave it to you.

**Comment by Mugen87** [comment]
> Do you want me to squash the fixup commits and force-push the changes?

That would be great. The merge option `Squash and merge` is not enabled for this repo.

**Comment by Peque** [comment]
@Mugen87 Done. :blush:

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**
